### PR TITLE
8287906: Rewrite of GitHub Actions (GHA) sanity tests

### PR DESCRIPTION
**Pull request by gdams**
Backports the GitHub Actions framework rewrite in preparation for https://bugs.openjdk.org/browse/JDK-8287906

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8287906](https://bugs.openjdk.org/browse/JDK-8287906): Rewrite of GitHub Actions (GHA) sanity tests

### Reviewers
 * [Christoph Langer](https://openjdk.org/census#clanger) (@RealCLanger - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk17u-dev pull/539/head:pull/539` \
`$ git checkout pull/539`

Update a local copy of the PR: \
`$ git checkout pull/539` \
`$ git pull https://git.openjdk.org/jdk17u-dev pull/539/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 539`

View PR using the GUI difftool: \
`$ git pr show -t 539`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk17u-dev/pull/539.diff">https://git.openjdk.org/jdk17u-dev/pull/539.diff</a>

</details>

### CODE REVIEW

**Comment by gdams** [comment]
Closing until the MYSYS backports are complete

**Comment by gdams** [comment]
Windows test failure will be fixed by https://github.com/openjdk/jdk17u-dev/pull/546

File: make/conf/github-actions.conf [hunk 1]
```diff
@@ -23,21 +23,19 @@
...
 
-BOOT_JDK_VERSION=17
-JTREG_VERSION=6
-JTREG_BUILD=1
 GTEST_VERSION=1.8.1
+JTREG_VERSION=6.1+2
 
-LINUX_X64_BOOT_JDK_FILENAME=openjdk-17.0.2_linux-x64_bin.tar.gz
+LINUX_X64_BOOT_JDK_EXT=tar.gz
 LINUX_X64_BOOT_JDK_URL=https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.2%2B8/OpenJDK17U-jdk_x64_linux_hotspot_17.0.2_8.tar.gz
```
Comment (RealCLanger) [inline make/conf/github-actions.conf:32]: Maybe you want to update this to 17.0.3?

**Review by RealCLanger — APPROVED** [review]
This looks good and testing is green. There was [JDK-8274980](https://bugs.openjdk.org/browse/JDK-8274980) which introduced some changes to the GHA workflow before, that is dropping the --with-version-build=0 configure value but that shouldn't be an issue. So good to go, modulo a bump of the boot JDK patch level.

**Comment by gdams** [comment]
@RealCLanger can you add the fix request label please?

**Comment by RealCLanger** [comment]
> @RealCLanger can you add the fix request label please?

Done.

**Pull request closed without merging**